docs/Music Macro Language.md

```diff
@@ -0,0 +1,989 @@
+delay 100        # Plays for 100 ticks, or approximately 2 beats
+```
+
+**rtranspose** *transposition*
+
+**Description:** Applies a relative transposition for notes played on all channels on this sequence player.  Subsequent calls of this command will continue to add the provided transposition to the current transposition on the sequence player.  By default, the sequence player's transposition is set to 0.
+
+The player's transposition is stored as a 16-bit signed integer.  If you apply a transposition that's larger than 32767, this will result in an integer overflow.  In practice, this should not be an issue.
+
+Sequence transpositions are ignored by drum and sound effects.
+
+**Parameters:**
+* `transposition` [integer: -128-127] - The relative change to apply to the current transposition
+
+**Example:**
+```
+transpose 10   # Transposes the notes played up by 10 semitones
+delay 50
+rtranspose -5  # Changes the transposition to 5 semitones above base
+delay 50
+```
+
+**transpose** *transposition*
+
+**Description:** Applies a transposition for notes played on all channels on this sequence player.  Unlike `rtranspose`, this forces the transposition to the number of semitones.  By default, the sequence player's transposition is set to 0.
+
+The player's transposition is stored as a 16-bit signed integer.  If you apply a transposition that's larger than 32767, this will result in an integer overflow.  In practice, this should not be an issue.
+
+Sequence transpositions are ignored by drum and sound effects.
+
+**Parameters:**
+* `transposition` [integer: -128-127] - The number of semitones to transpose by
+
+**Example:**
+```
+transpose 10   # Transposes the notes played up by 10 semitones
+delay 50
+```
+
+**freenotelist**
+
+**Description:** Frees the notes reserved on the sequence player and returns them back to the global note pool.
+
+It's recommended to avoid using this command while audio is playing for the sequence, as it can cause undefined behavior.
+
+**Parameters:** None.
+
+**Example:**
+```
+ldchan 1, chan1
+ldchan 9, chan9
+delay 1500
+freenotelist      # Frees any reserved notes on the sequence player.
+```
+
+**allocnotelist** *count*
+
+**Description:** Frees any reserved notes assigned to the sequence player, then reserves the number of notes requested for the sequence player.  The notes in this pool can then be used by channels and note layers when needed, depending on the note allocation policy set.
+
+It's recommended to avoid using this command while audio is playing on the sequence, as this can result in undefined behavior.
+
+In general, it is not necessary to use this command, as by default the sequence player will use any free notes in the global note pool.  This command also is not useful for addressing issues with notes cutting out on channels in a sequence, as this applies to the sequence player itself.  Howver, it can be useful for preventing other sequences from stealing notes from this sequence.
```

Comment:
```suggestion
In general, it is not necessary to use this command, as by default the sequence player will use any free notes in the global note pool.  This command also is not useful for addressing issues with notes cutting out on channels in a sequence, as this applies to the sequence player itself.  However, it can be useful for preventing other sequences from stealing notes from this sequence.
```

Reply:
Applied in #1509.